### ISSUE TITLE
Screenreaders don't read video subtitles automatically

### ISSUE THREAD

**Issue by RastislavKish** [videojs/video.js#7430]
## Description

Screenreaders don't read subtitles of videos automatically, since the element holding their text doesn't have aria-live attribute enabled.

## Steps to reproduce

1. Open a video with subtitles.
2. Launch a full-fledged screenreader, like [Orca](https://gitlab.gnome.org/GNOME/orca) for Linux, [NVDA](https://www.nvaccess.org/) for Windows or VoiceOver for Mac.
3. Turn video subtitles on.

### Expected

The screenreader automatically reads the subtitles as they're appearing on the screen.

### Actual

The screenreader doesn't report anything, the user has to manually check whether new text has been shown.

## Additional Information

Part of [FreeTube accessibility improvements PR](https://github.com/FreeTubeApp/FreeTube/pull/1525) by @jasonhenriquez.

#### videojs

7.14.3

#### browsers

Tested under Chromium (Electron), though the problem is not browser-related.

#### OSes

Tested on Ubuntu mate 20.04 64-bit and Windows 10 21H1 64-bit, though the problem is not platform related.

**Comment by gkatsev** [comment]
This is sort of by design to match how browsers work currently. Native captions aren't accessible by screen readers. When a descriptions track is enabled, on non-safari browsers, we make it available for Screen Readers to read.
There's certainly an argument to be made that captions should be accessible to SRs.
@OwenEdwards do you have any insights here?

**Comment by gkatsev** [comment]
I guess there's some discussion related to this topic here https://github.com/videojs/video.js/issues/3554 previously.

**Comment by jasonhenriquez** [comment]
A reasonable compromise between the points mentioned in #3554 and the desired utility sought here would be providing an option to enable this behavior.

**Comment by OwenEdwards** [comment]
Why is "The screenreader automatically reads the subtitles as they're appearing on the screen." the expected behavior? There is no standard expected behavior defined for this, and as @gkatsev says, the behavior of Video.js with Captions and Subtitles matches the native behavior of browsers when playing video with Captions and/or Subtitles.

**Comment by RastislavKish** [comment]
@gkatsev @OwenEdwards I am a screenreader user myself. The conclusion of #3554 is incomplete, as it doesn't take into account the cases where subtitles are used for translation purposes or annotate a video.
Currently, one needs to actively interact with a site to find out whether new text has appeared, what is inconvenient and distractful.

I have no direct experiences with development using VideoJS, so I'm not sure how exactly are the subtitles handled.
But in general:
* If subtitles are off by default and a user needs to turn them on manually, aria-live should be the default state.
* if this is upon programmers to decide, there should be a way for them to specify whether the subtitles should be live or not, as @jasonhenriquez suggests (such option should probably be present anyway).

**Comment by OwenEdwards** [comment]
@RastislavKish the problem is that other screen reader users may complain if video.js changes its behavior by putting captions in an `aria-live` region, especially for existing videos where a captions track defaults to enabled.

Even if the choice to have `aria-live` applied to captions were up to the web page developer, this feature could be misused and cause complaints from other screen reader users (because adding `aria-live` doesn't impact anyone _except_ screen reader users, so developers can be careless with applying it).

The only implementation that would make sense from a user perspective would be for the _user_ to be able to select whether the captions or subtitles are rendered in an `aria-live` region or not, in the same way that users can specify their own preferences for color, font, background, etc. of how captions and subtitles are rendered visually. So that would need to be a setting in the `textTrackSettings` modal dialog.

**Comment by RastislavKish** [comment]
@OwenEdwards The setting option sounds great. This way, everyone could select the preferred behavior according to their personal preference and needs.
Would be perhaps good to change this to a feature request, if we could go this way.

**Comment by gkatsev** [comment]
Yes, we can definitely make this a feature request for making captions screen reader accessible by user-request. Though, it's worth noting it'll only be available when native captions aren't being used (which is non-safari by default).
